### PR TITLE
Add configuration option to SinglePipe, to ignore 0-arity functions

### PR DESCRIPTION
**Pull request by begedin**
This is based on proposal-65 at https://github.com/rrrene/credo-proposals/issues/65

First time contributor to the repository, and I'm not really sure on what the conventions are, so any criticism is more than welcome.

### CODE REVIEW

File: lib/credo/check/readability/single_pipe.ex [hunk 1]
```diff
@@ -28,33 +29,77 @@ defmodule Credo.Check.Readability.SinglePipe do
...
+
+        my_function()
+        |> my_other_function()
+
+      ...could be considered easier to read than ...
+
+        my_other_function(my_function())
+
+      Because of this, the check can optionally ignore such cases (see configuration options).
+      """,
```
Comment (begedin) [inline lib/credo/check/readability/single_pipe.ex:45]: Not sure if explanation should be as verbose as this.

File: lib/credo/check/readability/single_pipe.ex [hunk 2]
```diff
@@ -28,33 +29,77 @@ defmodule Credo.Check.Readability.SinglePipe do
...
+
+
+
     {_continue, issues} =
-      Credo.Code.prewalk(source_file, &traverse(&1, &2, issue_meta), {true, []})
+      Credo.Code.prewalk(
+        source_file,
+        &traverse(&1, &2, issue_meta, allow_0_arity_functions),
+        {true, []}
+      )
```
Comment (begedin) [inline lib/credo/check/readability/single_pipe.ex:66]: Line got a bit long. Is this the desired format for splitting a long function call?

File: lib/credo/check/readability/single_pipe.ex [hunk 3]
```diff
@@ -28,33 +29,77 @@ defmodule Credo.Check.Readability.SinglePipe do
...
     issues
   end
 
   # TODO: consider for experimental check front-loader (ast)
-  defp traverse({:|>, _, [{:|>, _, _} | _]} = ast, {_, issues}, _) do
+  defp traverse({:|>, _, [{:|>, _, _} | _]} = ast, {_, issues}, _, _) do
+    {ast, {false, issues}}
+  end
+
+  # single pipe, with zero arity functions not allowed
```
Comment (begedin) [inline lib/credo/check/readability/single_pipe.ex:76]: This was a bit hard to parse, so I added an inline comment for every case

File: test/credo/check/readability/single_pipe_test.exs [hunk 4]
```diff
@@ -59,4 +59,71 @@ defmodule Credo.Check.Readability.SinglePipeTest do
...
+        bar |> foo()
+      end
+
+      def bar(_), do: nil
+    end
+    """
+    |> to_source_file()
+    |> run_check(@described_check, allow_0_arity_functions: true)
+    |> assert_issues()
+  end
```
Comment (begedin) [inline test/credo/check/readability/single_pipe_test.exs:127]: Not really fully clear if there's a scenario I'm missing here. Really not that experienced with AST. Open to ideas.

**Comment by rrrene** [comment]
Hi, thank you for this and sorry for the long wait. I made a couple of last minute touch ups and merged it.

One thing: Please use `mix format` when contributing to a project that has a `.formatter.exs` file :+1: (I don't mean this in a patronizing way, just sayin')

**Comment by begedin** [comment]
> Hi, thank you for this and sorry for the long wait. I made a couple of last minute touch ups and merged it.
> 
> One thing: Please use `mix format` when contributing to a project that has a `.formatter.exs` file 👍 (I don't mean this in a patronizing way, just sayin')

This somehow got lost in my notifications. I tend to use mix forma, and I thought I did here to. Might have had issues with it and it wasn't being called on save without my noticing. Apologies for that. Not taking it in a patronising way 👍
